stress/basic.go

```
@@ -0,0 +1,539 @@
+	}
+
+	if resp.StatusCode != http.StatusNoContent && resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf(string(body))
+	}
+
+	return resp, nil
+}
+
+// Send calls post and returns a response
+func (c *BasicClient) send(b []byte) response {
+	instanceURL := fmt.Sprintf("http://%v/write?db=%v&precision=%v", c.Address, c.Database, c.Precision)
+	t := NewTimer()
+
+	t.StartTimer()
+	resp, err := post(instanceURL, "application/x-www-form-urlencoded", bytes.NewBuffer(b))
+	t.StopTimer()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	r := response{
+		Resp:  resp,
+		Time:  time.Now(),
+		Timer: t,
+	}
+
+	return r
+}
+
+// BasicQuery implements the QueryGenerator interface
+type BasicQuery struct {
+	Template   Query `toml:"template"`
+	QueryCount int   `toml:"query_count"`
+	time       time.Time
+}
+
+// QueryGenerate returns a Query channel
+func (q *BasicQuery) QueryGenerate() <-chan Query {
+	c := make(chan Query, 0)
+
+	go func(chan Query) {
+		defer close(c)
+
+		for i := 0; i < q.QueryCount; i++ {
+			time.Sleep(10 * time.Millisecond)
+			c <- Query(fmt.Sprintf(string(q.Template), i))
+		}
+
+	}(c)
+
+	return c
+}
+
+// SetTime sets the internal state of time
+func (q *BasicQuery) SetTime(t time.Time) {
+	q.time = t
+
+	return
+}
+
+// BasicQueryClient implements the QueryClient interface
+type BasicQueryClient struct {
+	Address       string `toml:"address"`
+	Database      string `toml:"database"`
+	QueryInterval string `toml:"query_interval"`
+	Concurrency   int    `toml:"concurrency"`
+	client        client.Client
+}
+
+// Init initializes the InfluxDB client
+func (b *BasicQueryClient) Init() {
+	u, _ := url.Parse(fmt.Sprintf("http://%v", b.Address))
+	cl := client.NewClient(client.Config{
+		URL: u,
+	})
+
+	b.client = cl
+}
+
+// Query runs the query
+func (b *BasicQueryClient) Query(cmd Query, ts time.Time) response {
+	q := client.Query{
+		Command:  string(cmd),
+		Database: b.Database,
+	}
+
+	t := NewTimer()
+
+	t.StartTimer()
```

OK, that is better.

This timing pattern is all over the code. You might like to refactor it via embedding:

```
type TimingClient struct {
    Client
}

func (t *TimingClient) Query(...) time.Duration {
    t := NewTimer()
    t.client.Query(....)
    t.StopTimer()
    return t.Elapsed()
}
```
I also suggest you should not ignore the error codes.